docs/script_convert.py

```diff
@@ -0,0 +1,75 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import argparse
+import pathlib
+
+BASH = "# bash"
+BASH_IGNORE = "# bash-ignore"
+BASH_MULTILINE_COMMENT_START = ": '"
+BASH_MULTILINE_COMMENT_END = "'"
+
+
+def convert(args: argparse.Namespace):
+    src_path = pathlib.Path(args.script)
+    des_path = src_path.parent / f"{src_path.stem}.py"
+    with open(src_path, "r") as src_f:
+        with open(des_path, "w") as des_f:
+            line = src_f.readline()
+            bash_block = []
+            bash_detected = False
+            bash_ignore_detected = False
+            while line:
+                line = line.strip("\n").strip("\r")
+                if bash_detected:
+                    if line == BASH:
+                        # write the bash block to destination
+                        python_code = "# .. code-block:: bash\n#\n"
```

Comment:
should we run the bash script from this file and include the output? we could probably do that by injecting output markers into the script e.g. 

```
echo -e "\nSTART OF COMMAND 1 OUTPUT --->"
tvmc ...
echo -e "\n<--- START OF COMMAND 1 OUTPUT"
```

then we could take the `bash-ignore` commented areas,  extract them into a common prelude, and also include stuff like `set -eux` so that people write proper bash scripts. we could maybe do this by putting that into a `docs/bash_gallery_prelude.sh` and erroring in this script if we don't see a line like:

```
source docs/bash_gallery_prelude.sh || exit 2
```

might need to think about the path a little bit

Reply:
Is the idea to add the output of each command to generated python file and finally have it as part of generated HTML file?

Reply:
yeah exactly